lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -4971,22 +4986,32 @@ ByteCodeGenerator::GetLdSlotOp(Scope *scope, int envIndex, Js::RegSlot scopeLoca
             op = Js::OpCode::LdEnvSlot;
         }
     }
-    else if (!scope->GetCanMergeWithBodyScope() && this->currentScope == scope->GetFunc()->GetBodyScope())
-    {
-        // When param and body scopes are not merged the only place where you try to load the value from a param scoped symbol,
-        // while in body scope, is when we use its value to initialize the corresponding value in the body scope. That time we
-        // cannot use the normal LdLocalSlot as we have to use the param scope closure to load that field.
-        op = Js::OpCode::LdParamSlot;
-    }
     else if (scopeLocation != Js::Constants::NoRegister &&
              scopeLocation == funcInfo->frameSlotsRegister)
     {
-        op = Js::OpCode::LdLocalSlot;
+        if (!scope->GetCanMergeWithBodyScope() && this->currentScope == scope->GetFunc()->GetBodyScope())
```

Comment:
Do we really need this method to return LdParam[Slot,ObjSlot]? Isn't that opcode only for the case where you're emitting the initialization of the body scope symbols?